cbmc/patches/0001-Patch-MQTT-library-code-for-CBMC-Unsubscribe-proofs.patch

```diff
@@ -0,0 +1,87 @@
+From 9681dc68e81ea5afa75d310446199ae730ab9943 Mon Sep 17 00:00:00 2001
+From: Mark R Tuttle <mrtuttle@amazon.com>
+Date: Wed, 25 Mar 2020 14:09:26 +0000
+Subject: [PATCH] Patch MQTT library code for CBMC Unsubscribe proofs
+
+---
+ libraries/standard/common/include/iot_linear_containers.h |  5 +++++
+ libraries/standard/mqtt/include/iot_mqtt_protocol.h       |  2 ++
+ libraries/standard/mqtt/src/iot_mqtt_operation.c          | 10 ++++++++++
+ libraries/standard/mqtt/src/private/iot_mqtt_internal.h   |  2 ++
+ 4 files changed, 19 insertions(+)
+
+diff --git a/libraries/standard/common/include/iot_linear_containers.h b/libraries/standard/common/include/iot_linear_containers.h
+index 689ea5e..b9aec35 100644
+--- a/libraries/standard/common/include/iot_linear_containers.h
++++ b/libraries/standard/common/include/iot_linear_containers.h
+@@ -639,6 +639,11 @@ static inline IotLink_t * IotListDouble_FindFirstMatch( const IotListDouble_t *
+     /* Iterate through the list to search for matches. */
+     while( pCurrent != pList )
+     {
++#ifdef CBMC_UNSUBSCRIBEASYNC_PROOF
++      __CPROVER_assert(isMatch != NULL, "FindFirstMatch called with match predicate");
++      __CPROVER_assume(isMatch != NULL);
++#endif
```

Comment:
Is this for coverage purposes? Is that something that other proofs could benefit from? I'd update the macro to something more general, e.g., `CBMC_ISMATCH_COVERAGE`.

Reply:
Using CBMC_FINDFIRSTMATCH_MATCH_IS_NONNULL